docs/cql/tombstone-gc.rst

```diff
@@ -0,0 +1,109 @@
+=====================================
+Tombstone Garbage Collection (GC) 
+=====================================
+
+Overview
+----------
+When you delete some data from the database, the data record is not immediately 
+removed. Instead, it is marked with a tombstone - a data marker that prevents 
+the data from being returned as a query result. The data covered by the tombstone 
+and the tombstone itself are eventually removed after the compaction process 
+is performed.
+
+ScyllaDB allows you to manage how and when tombstones are removed, considering 
+that tombstones must be garbage-collected:
+
+* Within a reasonable time because a large number of tombstones impacts query 
+  response times and leads to performance degradation.
+* After a cluster-wide repair to ensure that all replicas are in sync. Otherwise, 
+  tombstones might be missing on some replica nodes, which may lead to 
+  the resurrection of deleted data.
+
+You can choose one of the following ways to garbage-collect tombstones:
+
+* Repair-based Tombstone GC (recommended) - tombstones are removed when a repair 
+  process has been performed.
+* Timeout-based Tombstone GC - tombstones are removed when they exceed the time 
+  specified with the ``gc_grace_seconds`` option.
+
+
+Repair-based Tombstone GC (recommended)
+-----------------------------------------
+
+When repair-based tombstone GC is enabled, tombstones are only removed after repair 
+is performed. Repair synchronizes data between nodes so that all replicas hold 
+the same data. In this way, repair-based tombstone GC prevents data resurrection.
```

Comment:
I'm not sure it's clear enough what this mode really does. I think the point is that repair-based tombstone gc will only delete tombstones that are "older" then the most recent repair, not deleting newer tombstones 
(by the way, the notion of what exactly is a tombstone's time is complicated, especially in the case of deletions that happened via TTL, I wonder if this is documented somewhere?).

@asias does this mode assume the timestamps being real time in some precision (I guess microseconds)? Is this assumption documented somewhere?

Reply:
How tombstone works and how the timestamp is used in the system are indeed complicated. I do not think we have user faced documents to explain the internals. There are two timestamp for a tombstone. One is the write time which uses api::timestamp. The other is the "arrive time" when the tombstone is received on the server which uses gc_clock::timestamp.